### ISSUE TITLE
Model.objects.create can't take an instance parameter

### ISSUE THREAD

**Issue by benjaminrigaud** [stphivos/django-mock-queries#50]
Hello,
I have been trying to use django-mock-queries with FactoryBoy and I had an issue when creating a set of instances via a DjangoModelFactory using a SubFactory

I made a repo to reproduce the issue:

https://github.com/benjaminrigaud/django-mock-queries-and-factory-boy
```python
======================================================================
ERROR: test_mock (issue.main.tests.test_mock_factory_boy.FactoryBoyTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/me/venvs/django-mock-queries-and-factory-boy/local/lib/python2.7/site-packages/mock/mock.py", line 1305, in patched
    return func(*args, **keywargs)
  File "/home/me/venvs/django-mock-queries-and-factory-boy/local/lib/python2.7/site-packages/django_mock_queries/mocks.py", line 281, in absorb_mocks
    return target(test_case)
  File "/home/me/venvs/django-mock-queries-and-factory-boy/local/lib/python2.7/site-packages/mock/mock.py", line 1305, in patched
    return func(*args, **keywargs)
  File "/home/me/venvs/django-mock-queries-and-factory-boy/local/lib/python2.7/site-packages/django_mock_queries/mocks.py", line 281, in absorb_mocks
    return target(test_case)
  File "/home/me/projects/django-mock-queries-and-factory-boy/issue/main/tests/test_mock_factory_boy.py", line 29, in test_mock
    b2 = BFactoryWithSub()
  File "/home/me/venvs/django-mock-queries-and-factory-boy/local/lib/python2.7/site-packages/factory/base.py", line 69, in __call__
    return cls.create(**kwargs)
  File "/home/me/venvs/django-mock-queries-and-factory-boy/local/lib/python2.7/site-packages/factory/base.py", line 623, in create
    return cls._generate(True, attrs)
  File "/home/me/venvs/django-mock-queries-and-factory-boy/local/lib/python2.7/site-packages/factory/base.py", line 548, in _generate
    obj = cls._prepare(create, **attrs)
  File "/home/me/venvs/django-mock-queries-and-factory-boy/local/lib/python2.7/site-packages/factory/base.py", line 523, in _prepare
    return cls._create(model_class, *args, **kwargs)
  File "/home/me/venvs/django-mock-queries-and-factory-boy/local/lib/python2.7/site-packages/factory/django.py", line 181, in _create
    return manager.create(*args, **kwargs)
  File "/home/me/venvs/django-mock-queries-and-factory-boy/local/lib/python2.7/site-packages/mock/mock.py", line 1062, in __call__
    return _mock_self._mock_call(*args, **kwargs)
  File "/home/me/venvs/django-mock-queries-and-factory-boy/local/lib/python2.7/site-packages/mock/mock.py", line 1128, in _mock_call
    ret_val = effect(*args, **kwargs)
  File "/home/me/venvs/django-mock-queries-and-factory-boy/local/lib/python2.7/site-packages/django_mock_queries/query.py", line 199, in create
    raise ValueError('{} is an invalid keyword argument for this function'.format(k))
ValueError: a is an invalid keyword argument for this function

----------------------------------------------------------------------

```
I am digging a bit to understand the limitations.

**Comment by benjaminrigaud** [comment]
It doesn't have anything to do with FactoryBoy: mocked create can't handle instances parameters:
```python
    @mocked_relations(ModelA)
    @mocked_relations(ModelB)
    def test_mock(self):
        a = ModelA.objects.create()
        b = ModelB.objects.create(a=a)
```
fails the same way.

**Comment by szykin** [comment]
Yeah, It's because we check for field's attname, `make_id` instead of field's name `make`.

```python
if k not in [f.attname for f in mock_set.model._meta.concrete_fields]:
    raise ValueError('{} is an invalid keyword argument for this function'.format(k))
```
Since we do not explicitly handle pk's in django-mock-queries, I propose to change it to check for ```f.name``` instead.

**Comment by benjaminrigaud** [comment]
Changing f.attname by f.name in create fixed this issue.

What do you mean by "we do not explicitly handle pk's in django-mock-queries"? What are the known limitations?

**Comment by stphivos** [comment]
Hi, I just created a fix #51 that handles both cases. I will merge it as soon as travis checks pass.

@szykin I used `find_field_names` in `utils.py` to validate which field names are allowed. Let's use that everywhere so we don't have to worry anymore and also to fix one place if needed in the future!
